apps/air-discount-scheme/api/src/app/modules/discount/discount.resolver.ts

```diff
@@ -31,7 +31,11 @@ export class DiscountResolver {
           let discount: TDiscount = await backendApi.getDiscount(
             relation.nationalId,
           )
-          if (!discount || discount.expiresIn <= TWO_HOURS) {
+          if (
+            !discount ||
+            discount.expiresIn <= TWO_HOURS ||
+            discount.discountCode.length !== 8
```

Comment:
What is this?

Reply:
Ah, good catch.
I currently make the `discountCode` equal to an empty string (and then force regeneration with the length check). 
The cache lives differently now that it has two types of discount codes in it but I have since fixed an issue with `connectionDiscountCodes` being regenerated every time the regular `discountCode` is used. I forgot to change this "hack" as well.

What should be done instead is simply regenerate a `discountCode`, make that code point to the cache and delete the old `discountCode` reference.

Reference:
```
type ConnectionDiscountCodes = {
  code: string
  flightId: string
  flightDesc: string
  validUntil: string
}[]

interface Discount {
  discountCode: string
  connectionDiscountCodes: ConnectionDiscountCodes
  nationalId: string
  expiresIn: number
}
```

Reply:
Rewrote this retriggering 'hack' away. Retrigger is now done by invalidating cache pointers